app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -104,6 +108,92 @@ export const exportedForTest = {
 };
 
 const PostHandler = <TBase extends Constructor<ServerDataOperatorBase>>(superclass: TBase) => class extends superclass {
+    /**
+     * handleScheduledPosts: Handler responsible for the Create/Update operations occurring the SchedulePost table from the 'Server' schema
+     * @param {HandleScheduledPostsArgs} ScheduledPostsArgs
+     * @param {boolean} draftsArgs.prepareRecordsOnly
+     * @returns {Promise<Model[]>}
+     */
+    handleScheduledPosts = async ({actionType, scheduledPosts, prepareRecordsOnly}: HandleScheduledPostsArgs): Promise<ScheduledPostModel[]> => {
+        const database: Database = this.database;
+
+        if (!scheduledPosts?.length && actionType === ActionType.SCHEDULED_POSTS.RECEIVED_ALL_SCHEDULED_POSTS) {
+            await database.write(async () => {
+                const scheduledPostCollection = database.get<ScheduledPostModel>(SCHEDULED_POST);
+                await scheduledPostCollection.query().destroyAllPermanently();
+            });
+            return [];
```

Comment:
Same in every place you are returning the empty list. You should not return an empty list but all the modified models.